pkg/controller/migplan/registry_health.go

```diff
@@ -0,0 +1,150 @@
+package migplan
+
+import (
+	"context"
+	"github.com/konveyor/mig-controller/pkg/apis/migration/v1alpha1"
+	"github.com/konveyor/mig-controller/pkg/compat"
+	corev1 "k8s.io/api/core/v1"
+	k8sLabels "k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"time"
+)
+
+// A registry health checker which routinely checks whether registries are healthy or not
+type registryHealth struct {
+	hostClient client.Client
+	Interval   time.Duration
+	handler    handler.EventHandler
+	queue      workqueue.RateLimitingInterface
+	predicates []predicate.Predicate
+	podLabels  map[string]string
+}
+
+// Start the health checks
+func (r *registryHealth) Start(
+	handler handler.EventHandler,
+	queue workqueue.RateLimitingInterface,
+	predicates ...predicate.Predicate) error {
+
+	r.handler = handler
+	r.queue = queue
+	r.predicates = predicates
+	go r.run()
+
+	return nil
+}
+
+// Enqueue a reconcile request event for migplan
+func (r *registryHealth) enqueue(plan v1alpha1.MigPlan) {
+	event := event.GenericEvent{
+		Meta:   &plan.ObjectMeta,
+		Object: &plan,
+	}
+	for _, predicate := range r.predicates {
+		if !predicate.Generic(event) {
+			return
+		}
+	}
+
+	r.handler.Generic(event, r.queue)
+}
+
+//Run the health checks for registry pods
+func (r *registryHealth) run() {
+	//List all the migplans using the hostClient
+	//Now using srcClient for each migplan find the registry pods, if registry container is not ready then enqueue this migplan
+	//repeat all the above steps for all the plans for both clusters
+
+	for {
+		time.Sleep(r.Interval)
+		planList, err := v1alpha1.ListPlans(r.hostClient)
```

Comment:
@alaypatel07 I'm not seeing where we are using a LabelSelector here as we discussed. Is that in a follow-on PR?